plugins/link/dialogs/link.js

```diff
@@ -983,6 +1014,28 @@
 			}
 		};
 	} );
+
+	function validateTelNumber() {
+		var dialog = this.getDialog(),
+			editor = dialog._.editor,
+			regExp =  editor.config.linkTelNumberValidate_regExp,
+			msg = editor.config.linkTelNumberValidate_msg,
+			linkLang = editor.lang.link,
+			func = CKEDITOR.dialog.validate.notEmpty( linkLang.noTel ),
+			messageWhenEmpty = func.apply( this );;
+
+		if ( !dialog.getContentElement( 'info', 'linkType' ) || dialog.getValueOf( 'info', 'linkType' ) != 'tel' ) {
+			return true;
+		}
+
+		if ( typeof messageWhenEmpty === 'string' || !messageWhenEmpty ) {
+			return messageWhenEmpty;
+		}
+
+		if ( regExp && !regExp.test( this.getValue() ) ) {
```

Comment:
Note that we have built-in `CKEDITOR.dialog.validate.regex`, which could simplify the logic.